garak/detectors/leakreplay.py

```diff
@@ -18,6 +18,9 @@ def detect(self, attempt: Attempt, case_sensitive=False) -> List[float]:
         detector_results = []
         if "triggers" in attempt.notes:
             triggers = attempt.notes["triggers"]
+            if hasattr(self, 'translator'):
+                if self.translator is not None:
+                    triggers = self.translator.translate_prompts(triggers)
```

Comment:
this piece of code appears a lot. we should really avoid this, and use/add/inject a hook in the base class

Reply:
Sorry I don't understand how to implement it.
Could you share with me your example code?